scheds/rust/scx_bpfland/src/main.rs

```diff
@@ -99,11 +100,10 @@ fn cpus_to_cpumask(cpus: &Vec<usize>) -> String {
     }
 
     // Convert the byte vector to a hexadecimal string.
-    let hex_str: String = bitmask
-        .iter()
-        .rev()
-        .map(|byte| format!("{:02x}", byte))
-        .collect();
+    let hex_str: String = bitmask.iter().rev().fold(String::new(), |mut f, byte| {
+        let _ = write!(&mut f, "{:02x}", byte);
+        f
```

Comment:
I get that this is equivalent, but is this necessarily better?

Reply:
Not exactly the same, the change avoid a string allocation for each byte written.